release-scripts/check-publish-crates.sh

```diff
@@ -13,6 +13,24 @@ function cargo_publish_dry_run_all() {
     for crate in "${PUBLISH_CRATES[@]}" ; do
         echo "$crate";
         pushd crates/"$crate";
+        
+        # Validate that dependencies with multiple locations defined have
+        # a specific version pinned, to avoid failures at publish time (#2550)
+        deps_header=0
+        while read -r line; do
+            if [[ $line =~ ^\[dependencies\]$ ]]; then
```

Comment:
I think that we also need to handle sections like `[target.'cfg(not(target_arch = "wasm32"))'.dependencies]`

See:

https://github.com/infinyon/fluvio/blob/040655069aabb8ae358c6f5517a1cf0bbb25403f/crates/fluvio/Cargo.toml#L75

Reply:
Good point. I will adjust the regex to include those headers as well.